include/fdreadoutlibs/wib/RAWWIBTriggerPrimitiveProcessor.hpp

```diff
@@ -80,175 +80,37 @@ class RAWWIBTriggerPrimitiveProcessor
-             RAW_WIB_TP_SUBFRAME_SIZE);
-  }
-
-  int debug_nhits = m_payload_wrapper.rwtp->get_nhits();
-  uint16_t debug_padding = m_payload_wrapper.rwtp->get_padding_3(); // NOLINT
-  //TLOG() << "Number of hits, padding " << debug_nhits << ", " << debug_padding; 
-  TLOG_DEBUG(TLVL_WORK_STEPS) << "Number of hits, padding " << debug_nhits << ", " << debug_padding;
-
-  // raw recording 
-  tp_dump_frame();
-
-  // stitch TP hits
-  tp_stitch(std::move(&m_payload_wrapper));
-}
-
-void tp_unpack(frame_ptr fr)
-{
-  auto& srcbuffer = fr->get_raw_tp_frame_chunk();
-  int num_elem = fr->get_raw_tp_frame_chunksize();
-
-  if (num_elem == 0) {
-    TLOG_DEBUG(TLVL_WORK_STEPS) << "No raw WIB TP elements to read from buffer! ";
-    return;
-  }
-  if (num_elem % RAW_WIB_TP_SUBFRAME_SIZE != 0) {
-    TLOG_DEBUG(TLVL_WORK_STEPS) << "Raw WIB TP elements not multiple of subframe size (3)! ";
-    return;
-  }
-
-  int offset = 0;
-  while (offset <= num_elem) {
-
-    if (offset == num_elem) break;
-
-    // Create next TP frame
-    m_payload_wrapper.rwtp.reset(new types::RAW_WIB_TRIGGERPRIMITIVE_STRUCT::FrameType());
-    ::memcpy(static_cast<void*>(&m_payload_wrapper.rwtp->m_head),
-             static_cast<void*>(srcbuffer.data() + offset),
-             m_payload_wrapper.rwtp->get_header_size());
-    int nhits = m_payload_wrapper.rwtp->get_nhits();
-    uint16_t padding = m_payload_wrapper.rwtp->get_padding_3(); // NOLINT
-
-    if (padding == 48879) { // padding hex is BEEF, new TP format
-      TLOG_DEBUG(TLVL_WORK_STEPS) << "Unpack raw WIB TP frame version 2.";
-      unpack_tpframe_version_2(fr, nhits, offset);
-
-    } else { // old TP format
-      TLOG_DEBUG(TLVL_WORK_STEPS) << "Unpack raw WIB TP frame version 1.";
-      unpack_tpframe_version_1(fr, offset);
+      fwrite(&rwtp->m_head, 1, sizeof(dunedaq::detdataformats::wib::TpHeader),
+        m_output_file);
+      for (int i=0; i<rwtp->get_nhits(); i++) {
+        fwrite(&rwtp->m_blocks[i], 1,
+               sizeof(dunedaq::detdataformats::wib::TpData), m_output_file);
+      }
     }
-
-    offset += m_payload_wrapper.rwtp->get_frame_size();
   }
-}
 
-
-  void tp_stitch(frame_ptr fr)
+  void tp_stitch()
   {
-    auto& rwtp = fr->rwtp;
     m_frames++;
     int nhits = rwtp->get_nhits();
     uint64_t ts_0 = rwtp->get_timestamp(); // NOLINT
-    rwtp->set_timestamp(ts_0); // NOLINT
+    //rwtp->set_timestamp(ts_0); // NOLINT
     uint8_t m_channel_no = rwtp->m_head.m_wire_no; // NOLINT
     uint8_t m_fiber_no = rwtp->m_head.m_fiber_no; // NOLINT
     for (int i = 0; i < nhits; ++i) {
-      TLOG_DEBUG(TLVL_WORK_STEPS) << "Processing raw TP hit " << i << " out of " << nhits << " hits in this frame.";
-      //TLOG() << "Processing raw TP hit " << i << " out of " << nhits << " hits in this frame.";
+      TLOG_DEBUG(TLVL_WORK_STEPS) << "Processing raw TP hit " << i << " out of " << nhits << " hits in this   frame.";
```

Comment:
Work_steps will be a way too high verbosity level for TP stitching! This will flood output.